src/pages/search/advanced_search/LanguageFilter.tsx

```diff
@@ -0,0 +1,79 @@
+import { Box, Chip, FormControl, InputLabel, MenuItem, Select } from '@mui/material';
+import { useTranslation } from 'react-i18next';
+import { useHistory } from 'react-router';
+import { ResultParam } from '../../../api/searchApi';
+import i18n from '../../../translations/i18n';
+import { dataTestId } from '../../../utils/dataTestIds';
+import { languageOptions } from '../../registration/DescriptionPanel';
+
+export const LanguageFilter = () => {
+  const { t } = useTranslation();
+  const history = useHistory();
+  const searchParams = new URLSearchParams(history.location.search);
+  const languageParam = searchParams.get(ResultParam.PublicationLanguageShould)?.split(',') || [];
+
+  const handleChange = (selectedUris: string[] | null) => {
+    if (selectedUris && selectedUris.length > 0) {
+      const languages = selectedUris
+        .map((iso6393Code) => {
+          const selectedLanguage = languageOptions.find((language) => language.iso6393Code === iso6393Code);
+          return selectedLanguage ? selectedLanguage.iso6393Code : null;
+        })
+        .filter(Boolean);
+
+      if (languages.length > 0) {
+        searchParams.set(ResultParam.PublicationLanguageShould, languages.join(','));
+      } else {
+        searchParams.delete(ResultParam.PublicationLanguageShould);
+      }
+    } else {
+      searchParams.delete(ResultParam.PublicationLanguageShould);
+    }
+
+    history.push({ search: searchParams.toString() });
+  };
+
+  const handleDelete = (iso6393CodeToDelete?: string) => () => {
+    selectedLanguages.length > 1
+      ? handleChange(languageParam.filter((iso6393Code) => iso6393Code !== iso6393CodeToDelete))
+      : handleChange([]);
+  };
+
+  const selectedLanguages = languageParam
+    .map((iso6393Code) => languageOptions.find((language) => language.iso6393Code === iso6393Code))
+    .filter(Boolean);
```

Comment:
Denne har feil type ser jeg 
![bilde](https://github.com/BIBSYSDEV/NVA-Frontend/assets/6313468/93755334-f3e8-460c-8785-d7e44e8aaa01)

Filtreres jo med Boolean her da, så vi vet egentlig at den aldri er undefined, men vi må sikkert legge til en `as` her :(
```suggestion
    .filter(Boolean)  as Language[];
```